### DISCUSSION TITLE
Dynamic BLE addressing issue

### DISCUSSION THREAD

**Discussion by Semper-Viventem** [seemoo-lab/AirGuard#134] [Q&A]
How do you determine the fact that the device you found in the current scan is the same device that you found several scans ago? I can see that you match beacons by the device address, but SmartTags and AirTags use dynamic addressing and change addresses each 15-30 minutes.

**Comment by Sn0wfreezeDev** [comment]
We use several different approaches to match devices. The AirTag MAC address for example is derived by the advertised public key. Therefore, we can do matching based on the MAC address. The SmartTag has a privacyID, which we use to match it. Additionally, we have some heuristics to check if the SmartTag did change the privacyID and we actually see the same SmartTag after changing the ID.

**Reply by Semper-Viventem** [reply]
Oh sounds good. Could you help me to find the AirTag matching solution in the code? Probably I missed this but found only the `RiskLevelEvaluator` which takes beacons only by MAC address
